blocks/datasets/__init__.py

```diff
@@ -0,0 +1,233 @@
+from abc import ABCMeta, abstractmethod
+
+import numpy
+import six
+
+from blocks.utils import update_instance
+
+
+class Dataset(object):
+    """A dataset.
+
+    Dataset classes implement the interface to a particular dataset.
+
+    Attributes
+    ----------
+    sources : tuple of strings
+        The sources this dataset can provide. By default, these are
+        ``features`` and ``targets``.
+
+    Notes
+    -----
+    Datasets should only implement the interface; they are not expected to
+    perform the iteration over the actual data. As such, they are
+    stateless, and can be shared by different parts of the library
+    simultaneously.
+
+    """
+    __metaclass__ = ABCMeta
+    sources = ('features', 'targets')
```

Comment:
I do not think we should specify default source names here. Let users do their job of naming their data properly. In addition when I open a dataset class I want to see sources names and it is rather surprising for me that they are inherited.